app/services/search_algorithms.rb

```diff
@@ -0,0 +1,88 @@
+class SearchAlgorithms
+
+  def self.sync_search_vector
```

Comment:
So, on insert, we do a bit of extra work to create a big search vector for each discussion. This search vector includes weighted data for the discussion, its proposals, and its comments.

I haven't quite decided whether this is crazy or not. It happens in a background job, and the query is still really performant even on big discussions with tons of comments, but I wonder whether doing this every time anything gets made in the app will blow out our delayed job queue.

You can check out the weights I chose in `search_algorithms#discussion_vector`